applications/vanilla/modules/class.userdiscussionsmodule.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * User discussions module
+ *
+ * @copyright 2009-2017 Vanilla Forums Inc.
+ * @license http://www.opensource.org/licenses/gpl-2.0.php GNU GPL v2
+ * @package Vanilla
+ * @since 2.3
+ */
+
+/**
+ * Renders recent discussions started by a specified user.
+ */
+class UserDiscussionsModule extends Gdn_Module {
+
+    /** @var int Display limit. */
+    public $limit = 10;
+
+    /** @var  int The ID of the user for whom you are showing the comments. */
+    public $userID = null;
+
+    /**
+     *
+     *
+     * @throws Exception
+     */
+    public function __construct($sender) {
+        parent::__construct();
+        $this->_ApplicationFolder = 'vanilla';
+        $this->fireEvent('Init');
+        //If you are being executed from the profile controller, get the UserID.
+        if (strtolower(val('ControllerName', $sender)) === 'profilecontroller') {
+            $this->userID = $sender->User->UserID;
+        }
+    }
+
+    /**
+     * Get the data for the module.
+     *
+     * @param int|bool $limit Override the number of comments to display.
+     */
+    public function getData($limit = false) {
+        if (!$limit) {
+            $limit = $this->limit;
+        }
+        $userModel = new UserModel();
```

Comment:
Same as `UserCommentModule::getData`. Verify the presence of a valid user ID before proceeding.